File: EcoMind/vectara.py
import requests
import json

def vectara(api_key, sent_files):
    url = "https://api.vectara.io/v2/corpora/new/upload_file"

    payload={}

    # don't change the first part that says file for the below
    # can only put one file at a time
    files = [
      ('file', ('vectara_handbook', open(sent_files[0], 'rb'), 'application/octet-stream'))
    ]
    headers = {
      'Accept': 'application/json',
      'x-api-key': api_key
    }

    response_upload = requests.request("POST", url, headers=headers, data=payload, files=files)

    print(response_upload.text)


    url = "https://api.vectara.io/v2/query"

    payload = json.dumps({
      "query": "Am I allowed to bring pets to work?",
      "search": {
        "corpora": [
          {
            "custom_dimensions": {},
            "lexical_interpolation": 0,
            "semantics": "default",
            "corpus_key": "new"
          }
        ]
        },
      "stream_response": False
    })
    headers = {
      'Content-Type': 'application/json',
      'Accept': 'application/json',
      'x-api-key': 'zut_63JHduOA7XZXlewPHuQOD7YuamOctBXBaAeuQA'
    }

    response = requests.request("POST", url, headers=headers, data=payload)

    print(response.text)

    return response.text

File: EcoMind/vectara.js
// upload
document.getElementById('fileInput').addEventListener('change', async function() {
    const fileInput = document.getElementById('fileInput');
    const file = fileInput.files[0]; // Get the selected file
  
    if (!file) {
      alert('Please select a file first!');
      return;
    }
  
    const formData = new FormData();
    formData.append('file', file);
  
    try {
      const response = await fetch('https://api.vectara.io/v2/corpora/new/upload_file', {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'x-api-key': 'zut_63JHduOA7XZXlewPHuQOD7YuamOctBXBaAeuQA'
        },
        body: formData
      });
  
      if (!response.ok) {
        throw new Error('Network response was not ok ' + response.statusText);
      }
  
      const data = await response.json();
      console.log(data);
    } catch (error) {
      console.error('Error:', error);
    }
  });


// query

let data = JSON.stringify({
    "query": "what is climate change",
    "search": {
      "corpora": [
        {
          "custom_dimensions": {},
          "lexical_interpolation": 0,
          "semantics": "default",
          "corpus_key": "new"
        }
      ],
    },
    "generation": {
      "response_language": "auto",
      "citations": {
        "style": "none",
        "url_pattern": "https://vectara.com/documents/{doc.id}",
        "text_pattern": "{doc.title}"
      },
      "enable_factual_consistency_score": true
    },
    "stream_response": false
  });
  
let config = {
method: 'POST',
headers: { 
    'Content-Type': 'application/json', 
    'Accept': 'application/json', 
    'x-api-key': 'zut_63JHduOA7XZXlewPHuQOD7YuamOctBXBaAeuQA'
},
body: data
};

fetch('https://api.vectara.io/v2/query', config)
.then(response => response.json())
.then(data => {
    console.log(JSON.stringify(data));
})
.catch(error => {
    console.error(error);
});
